crates/router/src/core/locking.rs

```diff
@@ -0,0 +1,191 @@
+use actix_web::rt::time as actix_time;
+use error_stack::{IntoReport, ResultExt};
+use redis_interface as redis;
+use router_env::{instrument, logger, tracing};
+
+use super::errors::{self, RouterResult};
+use crate::routes;
+
+#[derive(Debug)]
+pub enum Lock {
+    Acquired(String),
+    Release(String),
+}
+
+pub struct LockingError {}
+
+#[derive(Clone, Debug)]
+pub enum LockStatus {
+    Acquired(String),
+    AlreadyLocked(String),
+    NotEnabled,
+    AcquireFailedRetriesExhausted(String),
+    Released(String),
+    ReleaseFailedRetriesExhausted(String),
+}
+
+#[derive(Debug)]
+pub enum ActionOnWaitTimeout {
+    Default,
+}
+
+#[derive(Debug)]
+pub struct LockingInput {
+    pub unique_locking_key: String,
+    pub api_identifier: String,
+    pub action_on_wait_timeout: ActionOnWaitTimeout,
+    pub merchant_id: String,
+}
+
+#[instrument(skip(state))]
+pub async fn get_key_and_lock_resource(
+    state: &routes::AppState,
+    locking_input: LockingInput,
+    _should_retry: bool,
+) -> RouterResult<LockStatus> {
+    let api_identifier = locking_input.api_identifier.as_str();
+
+    let is_locking_enabled_for_merchant = true;
```

Comment:
I don' think we will need this feature, merchant wise.

Reply:
Sure will remove it.